ulm-semantics/test/execution.md

```diff
@@ -199,6 +208,100 @@ module ULM-TEST-EXECUTION
         </k>
         <ulm-output> B:Bytes </ulm-output>
 
+    syntax Bool ::= isTestResult(K) [symbol(isTestResult), function]
+    rule isTestResult(_:K) => false  [owise]
+    rule isTestResult(evaluatedStorageKey(_:Bytes)) => true
+    rule isTestResult(evaluatedEventSignature(_:Int)) => true
+    rule isTestResult(encodedValues(_:Bytes)) => true
+    rule isTestResult(_:PtrValue) => true
+    rule isTestResult(#SetAccountStorageHook(_:Int, _:Int)) => true
+    rule isTestResult(#GetAccountStorageHook(_:Int)) => true
+    rule isTestResult(#Log3Hook(_:Int, _:Int, _:Int, _:Bytes)) => true
+
+    syntax StorageKey ::= storageKey(NonEmptyStatementsOrError)
+                        | storageKey(Expression)  [strict(1)]
+                        | storageKey(UlmExpression)  [strict(1)]
+                        | evaluatedStorageKey(Bytes)
+    rule storageKey(StorageName:String, Args:EncodeArgs)
+        => storageKey
+            ( concat
+                (   let buffer_id = :: bytes_hooks :: empty( .CallParamsList );
+                    .NonEmptyStatements
+                ,   codegenValuesEncoder
+                        ( buffer_id
+                        , (StorageName : str , encodeArgsToEncodeValues(Args))
+                        )
+                )
+            )
+    rule storageKey(Statements:NonEmptyStatements)
+        =>  storageKey({.InnerAttributes Statements buffer_id })
+    rule storageKey(ptrValue(_, u64(BytesId))) => storageKey(ulmBytesId(BytesId))
+    rule storageKey(ulmBytesValue(B:Bytes)) => evaluatedStorageKey(B)
+
+    syntax EventSignature ::= evaluatedEventSignature(Int)
+    rule eventSignature(Signature:String)
+        => evaluatedEventSignature(Bytes2Int(Keccak256raw(String2Bytes(Signature)), BE, Unsigned))
+
+    syntax EncodeValues ::= encodeValues(NonEmptyStatementsOrError)
+                          | encodeValues(Expression)  [strict(1)]
+                          | encodeValues(UlmExpression)  [strict(1)]
+                          | encodedValues(Bytes)
+    rule encodeValues(Args:EncodeArgs)
+        => encodeValues
+            ( concat
+                (   let buffer_id = :: bytes_hooks :: empty( .CallParamsList );
+                    .NonEmptyStatements
+                ,   codegenValuesEncoder
+                        ( buffer_id
+                        , encodeArgsToEncodeValues(Args)
+                        )
+                )
+            )
+    rule encodeValues(Statements:NonEmptyStatements)
+        =>  encodeValues({.InnerAttributes Statements buffer_id })
+    rule encodeValues(ptrValue(_, u64(BytesId))) => encodeValues(ulmBytesId(BytesId))
+    rule encodeValues(ulmBytesValue(B:Bytes)) => encodedValues(B)
+
+    rule SetAccountStorageHook(evaluatedStorageKey(B:Bytes), ptrValue(_, Value:Value))
+        => #SetAccountStorageHook
+                ( Bytes2Int(Keccak256raw(B), BE, Unsigned)
+                , valueToInteger(Value)
```

Comment:
It seems that, in the case `valueToInteger` returns error, the rules below on 270 and 274 will not cover all possible cases and evaluation will get stuck.

Reply:
Yes, it gets stuck, and valueToInteger evaluates to a term containing an error message somewhat explaining what went wrong. That's by design. I mean, is there a better option if the user asks to add a mock for a `SetAccountStorage(key, non-integer-value)` call?

Of course, one option would be to provide an `Int` directly to `SetAccountStorageHook ` instead of an `Expression`, and I'll do that if you think it's better, but this PR attempted to make tests more readable, and I'm assuming that using `Expression` helps.